### PR TITLE
feat: Add input support for EBU Teletext in MPEG-TS

### PR DESCRIPTION
**Pull request by marcusspangenberg**
- Add support for EBU Teletext input following Level 1.5 of the core specification ETSI EN 300 706 V1.2.1 (2003-04).
- Add support for webvtt in MP4 segments output.

Implements https://github.com/shaka-project/shaka-packager/issues/272

### CODE REVIEW

**Comment by Nixon197** [comment]
I tested Teletext implementation with live UDP input and it looks good.

File: packager/media/formats/mp2t/es_parser_teletext.cc [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+    }
+
+    return false;
+
+  } else if (packet_nr == 26) {
+    ParsePacket26(data_block);
+    return false;
+
+  } else if (packet_nr > 26) {
+    return false;
```
Comment (acris5) [inline packager/media/formats/mp2t/es_parser_teletext.cc:271]: packet28 has more language codec infos, so it's advisible to parse it and change teletext G0 charset accordingly

**Review by acris5 — APPROVED** [review]
Finally this feature works well

**Comment by acris5** [comment]
@marcusspangenberg I updated your code with cmake build changes in upstream, and also added additional languages support. Please, take a look on this patch (it is a diff from main)
[teletext.zip](https://github.com/shaka-project/shaka-packager/files/14052796/teletext.zip)

**Comment by joeyparrish** [comment]
Please merge with `main`, which has changed a lot in the recent `cmake` port.  Thanks!

**Comment by cosmin** [comment]
Moving to #1344

**Comment by cosmin** [comment]
@acris5 I moved this pull request (including rebasing it to latest main) to #1344. The additional language changes from your zip file look interesting as well, would you mind submitting that as a pull request (on top of #1344)?

**Pull request closed without merging**